### PR TITLE
Refactor bossbar code

### PR DESCRIPTION
**Pull request by Incompleteusern**

### CODE REVIEW

**Comment by magicus** [comment]
Can you explain with a sentence or two what you are doing. Did you find a better way to hide boss bars?

**Comment by magicus** [comment]
(Also, the term "refactor" is usually meant to imply that the structure of the code changes, but not the behavior. This more seems like a bug fix or an enhancement).

File: common/src/main/resources/wynntils.mixins.json [hunk 1]
```diff
@@ -1,6 +1,5 @@
 {
   "client": [
-    "MainMixin",
```
Comment (magicus) [inline common/src/main/resources/wynntils.mixins.json:3]: 👍 for sorting this!!!

**Comment by Incompleteusern** [comment]
> Can you explain with a sentence or two what you are doing. Did you find a better way to hide boss bars?

This removes many of the variables in `TrackedBar` by storing them inside a boss event, as well as replacing minecraft's boss event with one we have a reference to. This also gets rid of us duplicating progress info.

File: common/src/main/resources/wynntils.mixins.json (hunk 1 above)
Comment (Incompleteusern) [reply common/src/main/resources/wynntils.mixins.json:3]: That was done automatically haha

**Comment by Incompleteusern** [comment]
> (Also, the term "refactor" is usually meant to imply that the structure of the code changes, but not the behavior. This more seems like a bug fix or an enhancement).

Yeah, sorry

**Review by magicus — APPROVED** [review]

**Pull request merged**